docs/source/training.rst

```diff
@@ -0,0 +1,276 @@
+    train_dataset = glue_convert_examples_to_features(data['train'], tokenizer, max_length=128, task='mrpc')
+    valid_dataset = glue_convert_examples_to_features(data['validation'], tokenizer, max_length=128, task='mrpc')
+    train_dataset = train_dataset.shuffle(100).batch(32).repeat(2)
+    valid_dataset = valid_dataset.batch(64)
+
+    # Prepare training: Compile tf.keras model with optimizer, loss and learning rate schedule
+    optimizer = tf.keras.optimizers.Adam(learning_rate=3e-5)
+    loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    model.compile(optimizer=optimizer, loss=loss)
+
+    # Train and evaluate using tf.keras.Model.fit()
+    history = model.fit(train_dataset, epochs=2, steps_per_epoch=115,
+                        validation_data=valid_dataset, validation_steps=7)
+
+This example uses the built-in
+:func:`~transformers.data.processors.glue.glue_convert_examples_to_features`
+to tokenize MRPC and convert it to a TF ``Dataset`` object which can be
+passed to ``model.fit()``.
+
+With the tight interoperability between TensorFlow and PyTorch models, you
+can even save the model and then reload it as a PyTorch model:
+
+.. code-block:: python
+
+    model.save_pretrained('./save/')
+    pytorch_model = BertForSequenceClassification.from_pretrained('./save/', from_tf=True)
+
+
+.. _trainer:
+
+Trainer
+^^^^^^^
+
+We also provide a simple but feature-complete training and evaluation
+interface through :func:`~transformers.Trainer` and
+:func:`~transformers.TFTrainer`. You can train, fine-tune,
+and evaluate any Transformers model with a wide range of training options and
+with built-in features like logging, gradient accumulation, and mixed
+precision.
+
+Using the trainer requires first creating a model and defining a
+:func:`~transformers.data.DataCollator`, which is responsible for taking in a batch and preparing
+them to be fed into the model.  The Trainer will expect the collated batches
+to be a dictionary with keys corresponding to the kwargs that will be used to
+call our model. Below, we define a collator in the case where a batch
+consists of a list of dicts and returns a single dict with the examples
+concatenated.
+
+.. code-block:: python
+
+    from transformers import DataCollator
+
+    class MyCollator(DataCollator):
+        def collate_batch(self, batch):
+            input_ids = torch.stack([example['input_ids'] for example in batch])
+            attention_mask = torch.stack([example['attention_mask'] for example in batch])
+            labels = torch.stack([example['label'] for example in batch])
+
+            return {
+                'input_ids': input_ids,
+                'attention_mask': attention_mask,
+                'labels': labels
+            }
+
+Now, you can simply define a :func:`~transformers.Trainer` and
+:func:`~transformers.TrainingArguments` and the all
+of the underlying work is done for you.
+
+.. code-block:: python
```

Comment:
Could we use the same style as the usage page and have this block able to switch between PyTorch and TensorFlow? That would be super cool.